### PR TITLE
Correctly open uris containing spaces

### LINKED ISSUE #775: "Open in Terminal" menu option in Files does not work correctly when uri contains spaces
### What Happened?

If the "Open in Terminal" context menu option is used for a location containing spaces it opens a tab at the user Home directory instead.

### Steps to Reproduce

1. Open Files and navigate to a location whose uri contains spaces
2. Use the background context menu option "Open with ... Terminal"
3. Terminal opens with an extra tab at the Home folder

### Expected Behavior

Terminal opens with an extra tab at the requested location

### OS Version

8.x (Early Access)

### Software Version

Compiled from git

### Log Output

_No response_

### Hardware Info

_No response_

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #775

### CODE REVIEW

**Comment by teamcons** [comment]
Any ETA for merging this bugfix ? The PR seems stale

File: src/Widgets/TerminalWidget.vala [hunk 1]
```diff
@@ -776,21 +776,27 @@ namespace Terminal {
...
                     File file;
-
                     for (var i = 0; i < uris.length; i++) {
-                         file = File.new_for_uri (uris[i]);
-                         if ((path = file.get_path ()) != null) {
-                             uris[i] = Shell.quote (path) + " ";
+                        // Get sanitized unquoted path (Files sends uris that are escaped and quoted)
+                        // We do not want the `file://` scheme included
+                        // We assume dropped paths are absolute
+                        file = File.new_for_uri (Utils.sanitize_path (Shell.unquote (uris[i]), "", false));
```
Comment (ryonakano) [inline src/Widgets/TerminalWidget.vala:783]: This line causes the following warning on build:

```
../src/Widgets/TerminalWidget.vala:783.71-783.93: warning: unhandled error `GLib.ShellError'
  783 |                         file = File.new_for_uri (Utils.sanitize_path (Shell.unquote (uris[i]), "", false));
      |                                                                       ^~~~~~~~~~~~~~~~~~~~~~~              
```

File: src/Widgets/TerminalWidget.vala [hunk 2]
```diff
@@ -776,21 +776,27 @@ namespace Terminal {
                     var uris = selection_data.get_uris ();
                     string path;
                     File file;
-
                     for (var i = 0; i < uris.length; i++) {
-                         file = File.new_for_uri (uris[i]);
-                         if ((path = file.get_path ()) != null) {
-                             uris[i] = Shell.quote (path) + " ";
+                        // Get sanitized unquoted path (Files sends uris that are escaped and quoted)
```
Comment (ryonakano) [inline src/Widgets/TerminalWidget.vala:780]: > Files sends uris that are escaped and quoted

I'm not sure if this behavior is just about Files. I'm not sure about each implementation, but this branch works as expected if I open Terminal from Code, GNOME Files, and Dolphin, the file manager made for KDE.

(I'm concerning this comment sounds like as if this code only supports URIs passed from Files and does not support them passed from other apps)

File: src/Widgets/TerminalWidget.vala (hunk 1 above)
Comment (jeremypw) [reply src/Widgets/TerminalWidget.vala:783]: ShellError now handled

File: src/Widgets/TerminalWidget.vala (hunk 2 above)
Comment (jeremypw) [reply src/Widgets/TerminalWidget.vala]: I have amended the comment accordingly

**Comment by jeremypw** [comment]
@ryonakano Thanks for the review. I have now addressed your comments.

**Review by ryonakano — APPROVED** [review]

**Comment by teamcons** [comment]
_**YES**_
Thank you !!!!

**Pull request merged**